src/Data/Text/Utf8.hs

```diff
@@ -223,6 +319,17 @@ toLowerAscii cp
 lowerUtf8 :: Text -> Text
 lowerUtf8 = pack . map Char.toLower . unpack
 
+asciiCount :: Int
+asciiCount = 128
+
+{-# INLINE lowerCodePoint #-}
+-- | Lower-Case a UTF-8 codepoint.
+-- Uses 'toLowerAscii' for ASCII and 'Char.toLower' otherwise.
+lowerCodePoint :: Int -> Int
```

Comment:
Theoretically, we could always represent code points with `Char`s here, right? If GHC unboxes them as well as `Int`s, there should be no performance difference, but the code would be a bit clearer overall.

Reply:
You're right, though I would move this to another PR where we can apply this change to the Aho-Corasick code as well.